crates/networking/manager/src/gossipsub/handle_gossipsub.rs

```diff
@@ -78,8 +93,27 @@ pub async fn handle_gossipsub_message(message: Message, beacon_chain: &BeaconCha
                     signed_block.message.block_root()
                 );
 
-                if let Err(err) = beacon_chain.process_block(*signed_block).await {
-                    error!("Failed to process gossipsub beacon block: {err}");
+                let Ok(validation_result) =
+                    validate_beacon_block(beacon_chain, cached_db, &signed_block).await
+                else {
+                    return;
+                };
+
+                match validation_result {
+                    ValidationResult::Accept => {
+                        if let Err(err) = beacon_chain.process_block(*signed_block.clone()).await {
+                            error!("Failed to process gossipsub beacon block: {err}");
+                        }
+                        p2psender.send_gossip(GossipMessage {
+                            topic: GossipTopic::from_topic_hash(&message.topic)
+                                .expect("invalid topic hash"),
+                            data: signed_block.as_ssz_bytes(),
+                        });
+                    }
+                    ValidationResult::Ignore => warn!("Ignoring gossipsub beacon block"),
+                    ValidationResult::Reject => {
```

Comment:
Also would be nice to log the ignore/reject reasons because it'll be very hard to debug in runtime.